src/bindings/c/tests/ov_infer_request_test.cpp

```diff
@@ -341,6 +341,15 @@ TEST_P(ov_infer_request_test, infer_async_wait_for) {
     }
 }
 
+TEST_P(ov_infer_request_test, infer_async_wait_for_return_busy) {
```

Comment:
@riverlijunjie please, advice how to add test for `ov::Cancelled` - default test model is too small, and inference finishes earlier than cancel may be called

Reply:
maybe we can create a model with custom operation which sleeps for 500ms during its `evaluate` method

Reply:
Can we try to create multiple inferReqs to start_async() with single stream/thread and cancel them?

Reply:
Done